Cython/Utility/CConvert.pyx

```diff
@@ -130,3 +130,74 @@ cdef inline tuple {{to_tuple_cname}}({{base_type}} *v, Py_ssize_t length):
         Py_INCREF(value)
         PyTuple_SET_ITEM(t, i, value)
     return t
+
+#################### c_array_slice.to_py ####################
+
+cdef extern from *:
+    void Py_INCREF(object o)
+    void Py_DECREF(object o)
+    object PyErr_Format(exc, const char *format, ...)
+    int __Pyx_is_valid_index(Py_ssize_t i, Py_ssize_t limit)
+
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    tuple PyTuple_New(Py_ssize_t size)
+    void PyTuple_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    set PySet_New(void *)
+    int PySet_Add(object p, object o) except -1
+
+
+@cname("{{cname}}")
+cdef inline {{to_py_obj_type}} {{cname}}({{base_type}} *arr, Py_ssize_t arr_size, Py_ssize_t from_idx, Py_ssize_t to_idx, int wraparound, int boundscheck):
+    cdef Py_ssize_t i, length = 0
+
+    if wraparound:
+        if from_idx < 0: 
+            from_idx += arr_size
+
+        if to_idx < 0: 
+            to_idx += arr_size
+
+    if boundscheck:
+        if from_idx > to_idx:
+             from_idx = to_idx
```

Comment:
There is no guarantee that this makes `from_idx` and `to_idx` >= 0. What if `to_idx` is -2000 for an array of size 5? Then it's still negative after the wrap-around above, and so will be `from_idx`.

Reply:
I did 
```
from_idx = to_idx = 0
```